lib/util/event_util.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:convert';
+import 'package:Solon/util/user_util.dart';
+import 'package:http/http.dart' as http;
+import 'package:Solon/models/event.dart';
+import 'package:Solon/services/event_connect.dart';
+
+class EventUtil {
+  static Stream<List<Event>> _getList({
+    Function function,
+    String query,
+  }) async* {
+    http.Response response = await function(
+      query: query,
+    );
+    String prefLangCode = await UserUtil.getPrefLangCode();
+    List<Event> _events;
+    List collection;
+    if (response.statusCode == 200) {
+      collection = json.decode(response.body)['events'];
+      _events = collection
+          .map((json) => Event.fromJson(map: json, prefLangCode: prefLangCode))
+          .toList();
+    }
```

Comment:
I'm just going to use events as an example. Should we need to pass in prefLangCode from the utility or should the Event model itself deal with the prefLangCode by calling UserUtil.getPrefLangCode() in the model instead of in util?

Reply:
@raymondlee00

Reply:
import 'package:translator/translator.dart'; in lib/screens/forum/page.dart

Reply:
i think calling UserUtil.getPrefLangCode() in the model would be better. besides if we ever need the prefLangCode in events_util, just call UserUtil.getPrefLangCode() there. data passing thru widgets is confusing/messy; its better to get data from UserUtil instead.

Reply:
realized this is a bad idea because we have to await the prefLangCode and i dont want to have to deal with waiting on constructing models

Reply:
well actually i dont know i'll test it soon(?)

Reply:
go to sleep lol

Reply:
the sharedPrefs are almost always going to be accessible by the phone, since it's local storage, so awaiting it shouldn't be a problem, even if it's during event card construction.

Reply:
https://stackoverflow.com/questions/54549235/dart-await-on-constructor it doesnt look like async constructors are allowed

Reply:
try .then(() {})

Reply:
a .then block is no different from using async.